### DISCUSSION TITLE
Read a doxygen doc of a specific function

### DISCUSSION THREAD

**Discussion by tdegeus** [robotpy/cxxheaderparser#91] [General]
### Problem description

Thanks for the module!!

Consider
```cpp
namespace name {

/**
 * @brief My first function.
 * @param a This is a parameter.
 */
int foo(int a);

/**
 * This is a class
 */
class Bar
{
public:
    /**
     * @brief Some counter
     * @return integer
     */
    int count();
};

} // namespace name
```

What is the API to read the doxygen doc of `name::foo`, `name::Bar`, or `name::Bar::count`? 

In trying to get the first I already got lost. This is up to where I have gotten:
```python
data = parse_string(txt)
for i in data.namespace.namespaces["name"].functions:
    print(i.name)
```
whereby I get stuck as I don't know how to easily match the name. 

### Reproducible example code

```text
-
```

**Comment by virtuald** [comment]
The PQName has a [`format`](https://github.com/robotpy/cxxheaderparser/blob/29b71ab2edde7af3eff1b12687d72ff3c54296fd/cxxheaderparser/types.py#L176) function which turns the name into a consistent string.

**Comment by virtuald** [comment]
I will note that I definitely agree that using this for very simple things is still rather annoying. I'm definitely open to ideas for adding things to make it simpler to retrieve information. Or maybe someone needs to create a few examples to show how to do these things? Or add documentation.

**Comment by tdegeus** [comment]
I guess that if one knows the syntax, many things are workable. The structure of C++ can be 'hard' so that a perfect and extremely syntax is never easy. I will play around and consider submitting an example for review. For now I document this example:
```python
from cxxheaderparser.simple import parse_string

docstrings = {}

data = parse_string(txt)
for namespace in data.namespace.namespaces:
    for i in data.namespace.namespaces[namespace].functions:
        if i.doxygen:
            docstrings[f"{namespace}::{i.name.format()}"] = i.doxygen

    for i in data.namespace.namespaces[namespace].classes:
        class_name = i.class_decl.typename.format().split("class ")[1]
        docstrings[f"{namespace}::{class_name}"] = i.class_decl.doxygen
        for j in i.methods:
            if j.doxygen:
                docstrings[f"{namespace}::{class_name}::{j.name.format()}"] = j.doxygen
```

**Comment by virtuald** [comment]
The class name thing is a bit weird. Probably should add a `ClassDecl.get_name` which returns the typename segments only without 'class' or 'typename' (so just this line: https://github.com/robotpy/cxxheaderparser/blob/29b71ab2edde7af3eff1b12687d72ff3c54296fd/cxxheaderparser/types.py#L181)?
